jose-jwt/jwa/EcdhKeyManagementWinWithAesKeyWrap.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+
+namespace Jose
+{
+    public class EcdhKeyManagementWinWithAesKeyWrap : EcdhKeyManagementWin
+    {
+        private readonly AesKeyWrapManagement aesKW;
+        private readonly int keyLengthBits;
+        private readonly EcdhKeyManagementUnixWithAesKeyWrap ecdhKeyManagementUnixWithAesKeyWrap;
+
+        public EcdhKeyManagementWinWithAesKeyWrap(int keyLengthBits, AesKeyWrapManagement aesKw, EcdhKeyManagementUnixWithAesKeyWrap ecdhKeyManagementUnixWithAesKeyWrap) : base(false, ecdhKeyManagementUnixWithAesKeyWrap)
+        {
+            aesKW = aesKw;
+            this.keyLengthBits = keyLengthBits;
+            this.ecdhKeyManagementUnixWithAesKeyWrap = ecdhKeyManagementUnixWithAesKeyWrap;
+        }
+
+        public override byte[][] WrapNewKey(int cekSizeBits, object key, IDictionary<string, object> header)
+        {
+            if (key is ECDiffieHellman)
+            {
+                return ecdhKeyManagementUnixWithAesKeyWrap.WrapNewKey(cekSizeBits, key, header);
+            }
+            
+            var cek = Arrays.Random(cekSizeBits);
+
+            return new byte[][] { cek, this.WrapKey(cek, key, header) };
+        }
+
+        public override byte[] WrapKey(byte[] cek, object key, IDictionary<string, object> header)
+        {
+            if (key is ECDiffieHellman)
```

Comment:
aren't next call `base.WrapNewKey(keyLengthBits, key, header);` will take care of this?

Reply:
No, because the default windows implementation won't know how to handle ECDiffieHellman, so we delegate it to the newer implementation.

Reply:
Well spotted! Should be safe to remove.

Reply:
Actually no. I can't remove this. I tried and it broke many unit tests. Then I realised the call is base.WrapNewKey(keyLengthBits, key, header);
which is not the same as what we are doing when it's an ECDiffieHellman key (we call the WrapKey() then).